lib/elixir/test/elixir/kernel/expansion_test.exs

```diff
@@ -957,157 +957,6 @@ defmodule Kernel.ExpansionTest do
     end
   end
 
-  describe "cond" do
-    test "expands each clause" do
-      before_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            true -> x
-          end
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            true -> x()
-          end
-        end
-
-      assert expand(before_expansion) == after_expansion
-    end
-
-    test "does not share lexical scope between clauses" do
-      before_expansion =
-        quote do
-          cond do
-            1 -> import List
-            2 -> flatten([1, 2, 3])
-          end
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            1 -> :"Elixir.List"
-            2 -> flatten([1, 2, 3])
-          end
-        end
-
-      assert expand(before_expansion) == after_expansion
-    end
-
-    test "does not leaks vars on head" do
-      before_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            y = 2 -> y
-          end
-
-          :erlang.+(x, y)
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            y = 2 -> y
-          end
-
-          :erlang.+(x(), y())
-        end
-
-      assert expand(before_expansion) == after_expansion
-    end
-
-    test "does not leak vars" do
-      before_expansion =
-        quote do
-          cond do
-            1 -> x = 1
-            2 -> y = 2
-          end
-
-          :erlang.+(x, y)
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            1 -> x = 1
-            2 -> y = 2
-          end
-
-          :erlang.+(x(), y())
-        end
-
-      assert expand(before_expansion) == after_expansion
```

Comment:
Similar the tests that checks if variable leak between clauses are very important as well.